robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents Report Portal in ReportPortal
+
+    This holds the properties and function to interact with the Report Portal
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
```

Comment:
I'd add `self.verify_ssl` attribute here and use its value with `requests` methods across the code

Reply:
Here, I am initiating only the parameters which are dynamic using config file.

Also:
`req = requests.get(url='', parameters={}, verify_ssl=False)` looks much better than `req = requests.get(url='', parameters={}, verify_ssl=self.verify_ssl)`.

I don't want to unnecessarily set the variables.

Reply:
@rplevka valid suggestion for sure, but something we can easily iterate on with this design.